package.json

```diff
@@ -29,15 +29,15 @@
     "url": "https://github.com/Pomax/react-onclickoutside/issues"
   },
   "scripts": {
-    "build": "npm run build:es && npm run build:commonjs",
+    "build": "run-p build:*",
     "build:es": "cross-env BABEL_ENV=es babel src -d es",
     "build:commonjs": "cross-env BABEL_ENV=commonjs babel src -d lib",
     "lint": "eslint src/*.js ./test",
     "test": "run-s test:**",
-    "test:basic": "run-s lint build",
+    "test:basic": "run-s lint build:commonjs",
     "test:karma": "karma start test/karma.conf.js --single-run",
     "test:nodom": "mocha test/no-dom-test.js",
-    "precommit": "npm test && lint-staged",
+    "precommit": "lint-staged && npm test",
```

Comment:
"test" is your general runner for everything from linting to unit testing to coverage, etc, so removing linting from `test` is not really a good idea. However, whatever changes `npm test` introduces due to the build step updates files that are probably already in the added changeset, so putting lint-staged last checks against the updated files, which is better than checking first and then rebuilding.